### PR TITLE
Add a bearer auth feature for http-client

### PR DESCRIPTION
**Pull request by Savrov**
**Subsystem**
Http-Client

**Motivation**
Add support of Bearer tokens to the client-side. (Issue #1967 )

**Solution**
Added a BearerAuthProvider with an ability to load and refresh tokens.

### CODE REVIEW

File: ktor-client/ktor-client-features/ktor-client-auth/common/src/io/ktor/client/features/auth/providers/BearerAuthProvider.kt [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+fun Auth.bearer(block: BearerAuthConfig.() -> Unit) {
+    with(BearerAuthConfig().apply(block)) {
+        providers.add(BearerAuthProvider(refreshTokenFun, loadTokensFun, true, realm))
+    }
+}
+
+data class Tokens(
+    val accessToken: String,
+    val refreshToken: String,
+    val type: String
```
Comment (Ololoshechkin) [inline ktor-client/ktor-client-features/ktor-client-auth/common/src/io/ktor/client/features/auth/providers/BearerAuthProvider.kt:24]: Why do we need type here?
If it's always "Bearer" maybe it worth renaming the class to `BearerTokens` and removing the `type` param?

Comment (Savrov) [reply ktor-client/ktor-client-features/ktor-client-auth/common/src/io/ktor/client/features/auth/providers/BearerAuthProvider.kt:24]: Agree. Will fix it

**Comment by Savrov** [comment]
@Ololoshechkin , will be it merged? Or another fix needed?

**Comment by Ololoshechkin** [comment]
@Savrov Thanks! I will merge it to a separate branch for now and then we need to think about the API and before that we need to consider user stories. But from the backend perspective I am happy with your change, just we needs to design the API before adding to ktor.

**Comment by moallemi** [comment]
Any update about when you add it to ktor?

**Comment by ovitrif** [comment]
What else is needed to get this into the lib?

**Comment by grrigore** [comment]
Any updates?

**Pull request merged**